src/credentials/providers/envVarsCredentialsProvider.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Credentials, EnvironmentCredentials } from 'aws-sdk'
+import { EnvironmentVariables } from '../../shared/environmentVariables'
+import { CredentialType } from '../../shared/telemetry/telemetry.gen'
+import { getStringHash } from '../../shared/utilities/textUtilities'
+import { CredentialsId, CredentialsProvider, CredentialsProviderType } from './credentials'
+
+/**
+ * Credentials given by environment variables.
+ *
+ * @see CredentialsProviderType
+ */
+export class EnvVarsCredentialsProvider implements CredentialsProvider {
+    public static readonly AWS_ENV_VAR_PREFIX: string = 'AWS'
+
+    private credentials: EnvironmentCredentials | undefined
+
+    public async isAvailable(): Promise<boolean> {
+        const env = process.env as EnvironmentVariables
+        return env.AWS_ACCESS_KEY_ID && env.AWS_SECRET_ACCESS_KEY ? true : false
+    }
+
+    public getCredentialsId(): CredentialsId {
+        return {
+            credentialSource: this.getProviderType(),
+            credentialTypeId: 'variables',
+        }
+    }
+
+    public static getProviderType(): CredentialsProviderType {
+        return 'env'
+    }
+
+    public getProviderType(): CredentialsProviderType {
+        return EnvVarsCredentialsProvider.getProviderType()
+    }
+
+    public getTelemetryType(): CredentialType {
+        return 'other'
+    }
+
+    public getHashCode(): string {
+        return getStringHash(JSON.stringify(this.credentials))
+    }
+
+    public getDefaultRegion(): string | undefined {
+        const env = process.env as EnvironmentVariables
+        return env.AWS_REGION
+    }
+
+    public canAutoConnect(): boolean {
+        return true
+    }
+
+    public async getCredentials(): Promise<Credentials> {
+        if (!this.credentials) {
+            this.credentials = new EnvironmentCredentials(EnvVarsCredentialsProvider.AWS_ENV_VAR_PREFIX)
```

Comment:
Would this be able to refresh the credential if the env var is reset elsewhere? e.g. re-source from a ~/.bashrc or something?

Reply:
- I don't think there is an "environment changed" event we can listen to.
- We could poll to check for changes, on a timer...
- ... but I can't think of a use-case where some component that _we don't control_ would expect dependents to detect changes to the environment at runtime.
    - Note: re-sourcing ~/.bashrc cannot change the process environment.
- For cases that we control, we probably would explicitly re-initialize the credentials.